arch/arm/src/Makefile

```diff
@@ -138,6 +138,15 @@ $(KBIN): $(OBJS)
 board$(DELIM)libboard$(LIBEXT):
 	$(Q) $(MAKE) -C board libboard$(LIBEXT) EXTRAFLAGS="$(EXTRAFLAGS)"
 
+ifeq ($(CONFIG_LTO_FULL),y)
+  ifeq (,$(findstring("gcc", $(notdir $(LD)))))
+    LDSTARTGROUP := $(addprefix -Xlinker ,$(LDSTARTGROUP))
+    LDENDGROUP   := $(addprefix -Xlinker ,$(LDENDGROUP))
+    LDFLAGS      := $(addprefix -Xlinker ,$(LDFLAGS))
+    LDFLAGS      += $(CFLAGS)
+  endif
+endif
```

Comment:
I see
```
LDSTARTGROUP ?= --start-group
LDENDGROUP ?= --end-group
```
at line 92 and 93. Is it possible to change it to more agnostic variant like
```
ifeq ($(LD),$(CC))
  LDSTARTGROUP ?= -Wl,--start-group
  LDENDGROUP ?= -Wl,--end-group
else
  LDSTARTGROUP ?= --start-group
  LDENDGROUP ?= --end-group
endif
```
I'm not sure if that is reasonable, but maybe can be explored. I mean that linking with `ld` or `gcc` should not depend on `CONFIG_LTO_FULL=y` or `CONFIG_LTO_FULL=n`.

Reply:
Done